src/DeviceInput/deviceInputSystem.ts

```diff
@@ -0,0 +1,225 @@
+import { Engine } from '../Engines/engine';
+import { Nullable } from '../types';
+
+/**
+ * This class will take all inputs from Keyboard, Pointer, and
+ * any Gamepads and provide a polling system that all devices
+ * will use.  This class assumes that there will only be one
+ * pointer device and one keyboard.
+ */
+export class DeviceInputSystem {
+    // Private Members
+    private _inputs: Map<string, Array<number>> = new Map();
+    private _onDeviceConnected: (deviceName: string) => void = () => { };
+    private _onDeviceDisconnected: (deviceName: string) => void = () => { };

```

Comment:
Should these be [`Observable`](https://doc.babylonjs.com/api/classes/babylon.observable)s instead of callbacks?

Reply:
I'm not sure. Observables are heavier. This contract is not intended for Babylon users. It is a contract for internal use. That said, there still might be a good reason to use Observables even for internal use.